disnake/ui/action_row.py

```diff
@@ -82,6 +84,20 @@
     Sequence[Union["ActionRow[UIComponentT]", UIComponentT, Sequence[UIComponentT]]],
 ]
 
+# this is cursed
+ButtonCompatibleActionRow = TypeVar(
+    "ButtonCompatibleActionRow",
+    bound="Union[ActionRow[MessageUIComponent], ActionRow[WrappedComponent]]",
+)
+SelectCompatibleActionRow = TypeVar(
+    "SelectCompatibleActionRow",
+    bound="Union[ActionRow[MessageUIComponent],ActionRow[WrappedComponent]]",  # to add: ActionRow[ModalUIComponent]
+)
+TextInputCompatibleActionRow = TypeVar(
+    "TextInputCompatibleActionRow",
+    bound="Union[ActionRow[ModalUIComponent], ActionRow[WrappedComponent]]",
+)
```

Comment:
```suggestion
ButtonCompatibleActionRowT = TypeVar(
    "ButtonCompatibleActionRowT",
    bound="Union[ActionRow[MessageUIComponent], ActionRow[WrappedComponent]]",
)
SelectCompatibleActionRowT = TypeVar(
    "SelectCompatibleActionRowT",
    bound="Union[ActionRow[MessageUIComponent],ActionRow[WrappedComponent]]",  # to add: ActionRow[ModalUIComponent]
)
TextInputCompatibleActionRowT = TypeVar(
    "TextInputCompatibleActionRowT",
    bound="Union[ActionRow[ModalUIComponent], ActionRow[WrappedComponent]]",
)
```
usages of these variables need to be changed as well

Reply:
https://github.com/DisnakeDev/disnake/pull/740/commits/9e1fec56499ff94eb2a0e67857ef779e38f9af14